### PR TITLE
More Python3 fixes, and run tests with Tox instead of Circle overrides

### PR DESCRIPTION
**Pull request by mpetazzoni**
Some more Python 3 fixes, with same changes to how we run tests so we can actually run tests with both Python 2.x and Python 3.x and catch those Python 2/3 compatibility issues better in the future.

### CODE REVIEW

File: signalfx/ingest.py [hunk 1]
```diff
@@ -338,7 +338,8 @@ def _add_to_queue(self, metric_type, datapoint):
     def _batch_data(self, datapoints_list):
         datapoints = collections.defaultdict(list)
         for item in datapoints_list:
-            datapoints[item.keys()[0]].append(item[item.keys()[0]])
+            item_keys = list(item.keys())
```
Comment (tedoc2000) [inline signalfx/ingest.py]: item_key = item.keys()[0] ?

Comment (tedoc2000) [inline signalfx/ingest.py]: Oh look at that it is a set in python 3.. nm

**Comment by mpetazzoni** [comment]
Fixed the test for Python3, build is green now.

**Comment by tedoc2000** [comment]
LGTM do you want to squash or just merge as is?

**Comment by mpetazzoni** [comment]
Keep as-is, don't want the tox changes to collide with the Python 3 compat changes.

**Pull request merged**